src/cpp/src/tokenizer.cpp

```diff
@@ -707,6 +742,21 @@ void Tokenizer::set_chat_template(const std::string& chat_template) {
     m_pimpl->set_chat_template(chat_template);
 }
 
+Vocab Tokenizer::get_vocab() const {
```

Comment:
`m_vocab` is a `vector` but `Vocab` is a map. Is there a reason for such difference? I think it would be cleaner if the types matched.

Just curios why `Vocab` is the map str->int. Why not int->str or just a vector?